src/backend/distributed/utils/citus_copyfuncs.c

```diff
@@ -73,6 +74,20 @@ CitusSetTag(Node *node, int tag)
 	} \
 	while (0)
 
+#define COPY_STRING_LIST(fldname) \
+	char *curString = NULL; \
+	List *newList = NIL; \
+	foreach_ptr(curString, from->fldname) { \
+		char *newString = curString ? pstrdup(curString) : (char *) NULL; \
+		if (newList == NIL) { \
+			newList = list_make1(newString); \
+		} \
+		else { \
+			newList = lappend(newList, newString); \
+		} \
+	} \
+	newnode->fldname = newList;
```

Comment:
You could wrap the code in `do { } while(0)` for better scoping.

Reply:
umm, makes sense. For others: https://stackoverflow.com/questions/923822/whats-the-use-of-do-while0-when-we-define-a-macro